File: 알고리즘__문제/프로그래머스/프로그래머스_lever1/연습문제/정수내림차순으로배치하기.py
def solution(n):
    sort_num = []
    for i in sorted(str(n), reverse=True):
        sort_num.append(int(i))
    return "".join(map(str, sort_num))


n = 118372
print(solution(n))

# TypeError: sequence item 0: expected str instance, int found
# 에러가 발생하는 이유는 리스트안에 요소가 문자가 아닌 정수, 실수이기때문 따라서 요소를 문자로 바꿔주면 에러가 해결된다.

# sort와 sorted쓰는 방법은 다르다!
# ls.sort(reverse = True)

File: 자바스크립트풀이/programmers/정수내림차순으로배치하기.js
function solution(n) {
    return Number(String(n).split('').sort((x,y) => y - x).join(''));
}

solution(118372)

// 풀이 15분 소요

// function solution(n) {
//   return +String(n).split('').sort((p, c) => c - p).join('');
// }
// 앞에 +는 나머지 부분의 결과물로 나오는 문자열을 숫자로 바꾸는 것입니다.